pkg/oc/bootstrap/docker/up.go

```diff
@@ -242,12 +245,20 @@ func (c *ClusterUpConfig) Bind(flags *pflag.FlagSet) {
 	flags.StringArrayVarP(&c.Environment, "env", "e", c.Environment, "Specify a key-value pair for an environment variable to set on OpenShift container")
 	flags.BoolVar(&c.ShouldInstallMetrics, "metrics", false, "Install metrics (experimental)")
 	flags.BoolVar(&c.ShouldInstallLogging, "logging", false, "Install logging (experimental)")
-	flags.StringArrayVar(&c.AddComponents, "enable", c.AddComponents, "Install additional components.")
+	flags.StringSliceVar(&c.UserEnabledComponents, "enable", c.UserEnabledComponents, fmt.Sprintf(""+
+		"A list of components to enable.  '*' enables all on-by-default components, 'foo' enables the component "+
+		"named 'foo', '-foo' disables the component named 'foo'.\nAll components: %s\nDisabled-by-default components: %s",
+		strings.Join(knownComponents.List(), ", "), strings.Join(componentsDisabledByDefault.List(), ", ")))
 	flags.StringVar(&c.HTTPProxy, "http-proxy", "", "HTTP proxy to use for master and builds")
 	flags.StringVar(&c.HTTPSProxy, "https-proxy", "", "HTTPS proxy to use for master and builds")
 	flags.StringArrayVar(&c.NoProxy, "no-proxy", c.NoProxy, "List of hosts or subnets for which a proxy should not be used")
 }
 
+var (
+	knownComponents             = sets.NewString("service-catalog", "template-service-broker")
```

Comment:
how about registry, router and web-console? ;-)

Reply:
> how about registry, router and web-console? ;-)

Once your pull merges, we can migrate the rest.